servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java

```diff
@@ -0,0 +1,336 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.Constants;
+import tachyon.conf.TachyonConf;
+import tachyon.worker.block.meta.BlockMeta;
+import tachyon.worker.block.meta.StorageDir;
+import tachyon.worker.block.meta.StorageTier;
+import tachyon.worker.block.meta.TempBlockMeta;
+
+/**
+ * Manages the metadata of all blocks in managed space. This information is used by the
+ * TieredBlockStore, Allocator and Evictor.
+ * <p>
+ * This class is thread-safe and all operations on block metadata such as StorageTier, StorageDir
+ * should go through this class.
+ */
+public class BlockMetadataManager {
```

Comment:
Is there a defined metadata interface the users of this class should use? It seems like methods can be called on the metadata manager or on the tiers/dirs directly.